### PR TITLE
Check preconditions in patch.sh before using quilt

### PR DESCRIPTION
**Pull request by ed-**
Adds a few clauses zenrun's patch.sh to check that the patch and series
exist, and that there are any patches to rewind and then apply.

This should prevent apply-custom-patches from erroring out with the cryptic
message "No series file found".

### CODE REVIEW

File: bin/zenrun.d/patch.sh [hunk 1]
```diff
@@ -27,15 +27,33 @@ applyPatches() {
         return "$RC"
     fi
 
-    # First pop everything off (the image may or may not actually have the
-    # patches applied, so need to pop them first)
-    quilt pop -a
+    # Assure for patch directory and series file exist.
```
Comment (ed-) [inline bin/zenrun.d/patch.sh]: VERY USE OF ENGLISH ED

File: bin/zenrun.d/patch.sh [hunk 2]
```diff
@@ -27,15 +27,33 @@ applyPatches() {
...
     fi
-    
+
+    # Pop off old patches, if any
+    quilt top > /dev/null 2>&1
+    RC=$?
+    if [[ $RC == 0 ]]; then
+        quilt pop -a
+        RC=$?
+        if [[ $RC != ]];then
```
Comment (ed-) [inline bin/zenrun.d/patch.sh]: uhh

File: bin/zenrun.d/patch.sh [hunk 3]
```diff
@@ -27,15 +27,33 @@ applyPatches() {
...
+    # Assure that patch directory and series file exist.
+    if [[ ! -d patches ]]; then
+        mkdir patches
+    fi
+    touch patches/series
+
+    # Check for new patches
+    quilt unapplied > /dev/null 2>&1
     RC=$?
     if [[ $RC != 0 ]]; then
```
Comment (alapidas) [inline bin/zenrun.d/patch.sh]: Technically this gives us the status if the redirection, not the ```quilt unapplied``` command.  This issue exists in other places in this script as well.

File: bin/zenrun.d/patch.sh [hunk 4]
```diff
@@ -27,15 +27,33 @@ applyPatches() {
...
         return "$RC"
     fi
-    
+
+    # Pop off old patches, if any
+    quilt top > /dev/null 2>&1
+    RC=$?
+    if [[ $RC == 0 ]]; then
+        quilt pop -a
+        RC=$?
```
Comment (alapidas) [inline bin/zenrun.d/patch.sh]: You can just say ```if ! quilt pop -a;``` here

File: bin/zenrun.d/patch.sh (hunk 3 above)
Comment (ed-) [inline bin/zenrun.d/patch.sh]: Of course it does. I'll take care of it.

**Pull request merged**